maven/spec/dependabot/maven/file_fetcher_spec.rb

```diff
@@ -441,4 +441,161 @@
+          body: fixture("github", "parent_modules_different_artifact_pom.json"),
+          headers: { "content-type" => "application/json" }
+        )
+
+      stub_request(:get, File.join(url, "parent_modules_project/pom_with_parent_groupid/pom_with_parent/.mvn?ref=sha")).
+        with(headers: { "Authorization" => "token token" }).
+        to_return(
+          status: 404
+        )
+
+      stub_request(:get, File.join(url, "parent_modules_project/pom_with_no_parent/.mvn?ref=sha")).
+        with(headers: { "Authorization" => "token token" }).
+        to_return(
+          status: 404
+        )
+
+      stub_request(:get, File.join(url, "parent_modules_project/pom_with_implicit_parent_path/.mvn?ref=sha")).
+        with(headers: { "Authorization" => "token token" }).
+        to_return(
+          status: 404
+        )
+
+      stub_request(:get, File.join(url, "parent_modules_project/different_version_id/.mvn?ref=sha")).
+        with(headers: { "Authorization" => "token token" }).
+        to_return(
+          status: 404
+        )
+
+      stub_request(:get, File.join(url, "parent_modules_project/different_group_id/.mvn?ref=sha")).
+        with(headers: { "Authorization" => "token token" }).
+        to_return(
+          status: 404
+        )
+
+      stub_request(:get, File.join(url, "parent_modules_project/different_artifact_id/.mvn?ref=sha")).
+        with(headers: { "Authorization" => "token token" }).
+        to_return(
+          status: 404
+        )
+
+      stub_request(:get, File.join(url, ".mvn/extensions.xml?ref=sha")).
+        with(headers: { "Authorization" => "token token" }).
+        to_return(
+          status: 404
+        )
+    end
+
+    context "when asked to fetch a pom with a parent which has inherited group id" do
+      let(:directory) { "/parent_modules_project/pom_with_parent_groupid/pom_with_parent" }
+      it "fetches the relevant poms" do
+        expect(file_fetcher_instance.files.count).to eq(3)
+        expect(file_fetcher_instance.files.map(&:name)).
+          to match_array(%w(pom.xml ../pom_parent.xml ../../pom.xml))
+      end
+    end
+
+    context "when asked to fetch a pom with no parent defined" do
+      let(:directory) { "/parent_modules_project/pom_with_no_parent" }
+      it "fetches only this pom" do
+        expect(file_fetcher_instance.files.map(&:name)).to eq(%w(pom.xml))
+      end
+    end
+
+    context "when asked to fetch a pom with an implicit path to parent defined" do
+      let(:directory) { "/parent_modules_project/pom_with_implicit_parent_path" }
+      it "fetches only both this pom and its implicit parent" do
+        expect(file_fetcher_instance.files.count).to eq(2)
+        expect(file_fetcher_instance.files.map(&:name)).
+          to match_array(%w(pom.xml ../pom.xml))
+      end
+    end
+
+    context "when asked to fetch a pom which local parent has different versionId" do
+      let(:directory) { "/parent_modules_project/different_version_id" }
```

Comment:
How does maven behave here? It seems like a bad place to be for a repo, right? In any case, good with not messing with this kind of mismatch 👍.

Reply:
As a minor nit on top of this, I would it be `version` rather than `versionId` here?

Reply:
> How does maven behave here? It seems like a bad place to be for a repo, right? In any case, good with not messing with this kind of mismatch 👍.

Generally - it would throw a warning and then try to lookup in remote repositories - basically for any of these 3 cases when it fails to find anything 😃